applications/FluidDynamicsApplication/custom_conditions/low_mach_navier_stokes_wall_condition.cpp

```diff
@@ -0,0 +1,622 @@
+    const unsigned int x_pos = this->GetGeometry()[0].GetDofPosition(VELOCITY_X);
+    const unsigned int t_pos = this->GetGeometry()[0].GetDofPosition(TEMPERATURE);
+
+    unsigned int local_index = 0;
+    for (unsigned int i = 0; i < NumNodes; ++i) {
+        rResult[local_index++] = r_geometry[i].GetDof(PRESSURE, p_pos).EquationId();
+        rResult[local_index++] = r_geometry[i].GetDof(VELOCITY_X, x_pos).EquationId();
+        rResult[local_index++] = r_geometry[i].GetDof(VELOCITY_Y, x_pos+1).EquationId();
+        if constexpr (Dim == 3) {
+            rResult[local_index++] = r_geometry[i].GetDof(VELOCITY_Z, x_pos+2).EquationId();
+        }
+        rResult[local_index++] = r_geometry[i].GetDof(TEMPERATURE, t_pos).EquationId();
+    }
+}
+
+template<unsigned int TDim, unsigned int TNumNodes, class... TWallModel>
+void LowMachNavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::GetDofList(
+    DofsVectorType& rConditionDofList,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    const auto& r_geometry = this->GetGeometry();
+
+    if (rConditionDofList.size() != LocalSize) {
+         rConditionDofList.resize(LocalSize);
+    }
+
+    const unsigned int p_pos = this->GetGeometry()[0].GetDofPosition(PRESSURE);
+    const unsigned int x_pos = this->GetGeometry()[0].GetDofPosition(VELOCITY_X);
+    const unsigned int t_pos = this->GetGeometry()[0].GetDofPosition(TEMPERATURE);
+
+    unsigned int local_index = 0;
+    for (unsigned int i = 0; i < NumNodes; ++i) {
+        rConditionDofList[local_index++] = r_geometry[i].pGetDof(PRESSURE, p_pos);
+        rConditionDofList[local_index++] = r_geometry[i].pGetDof(VELOCITY_X, x_pos);
+        rConditionDofList[local_index++] = r_geometry[i].pGetDof(VELOCITY_Y, x_pos+1);
+        if constexpr (Dim == 3) {
+            rConditionDofList[local_index++] = r_geometry[i].pGetDof(VELOCITY_Z,x_pos+2);
+        }
+        rConditionDofList[local_index++] = r_geometry[i].pGetDof(TEMPERATURE, t_pos);
+    }
+}
+
+template<unsigned int TDim, unsigned int TNumNodes, class... TWallModel>
+void LowMachNavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize if needed
+    if (rLeftHandSideMatrix.size1() != LocalSize || rLeftHandSideMatrix.size2() != LocalSize) {
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+    }
+    if (rRightHandSideVector.size() != LocalSize) {
+        rRightHandSideVector.resize(LocalSize, false);
+    }
+
+    // LHS and RHS contributions initialization
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+
+    // // Struct to pass around the data
+    // ConditionDataStruct data;
+
+    // // Allocate memory needed
+    // array_1d<double,MatrixSize> rhs_gauss;
```

Comment:
this part is necessary?

Reply:
It is. I prefer it to leave it there for the eventual extension to unfitted mesh discretizations.